packages/xo-server-sdn-controller/src/index.js

```diff
@@ -638,11 +854,28 @@ class SDNController extends EventEmitter {
       })
     )
 
+    const crossPoolNetwork = find(this._crossPoolNetworks, crossPoolNetwork =>
+      crossPoolNetwork.networks.includes(network.$ref)
+    )
     if (newCenter === undefined) {
       log.error('No available host to elect new star-center', {
         network: network.name_label,
         pool: network.$pool.name_label,
       })
+
+      // Re-elect a cross-pool center if needed
+      if (
+        crossPoolNetwork !== undefined &&
+        crossPoolNetwork.poolCenter === network.$pool.$ref
+      ) {
+        const centerPoolNetwork = this._getPoolNetwork(
+          crossPoolNetwork.poolCenter,
+          crossPoolNetwork
+        )
+        delete centerPoolNetwork.starCenter
```

Comment:
Shouldn't this be in `_electNewPoolCenter`?

Reply:
No, I don't delete the `crossPoolNetwork` `poolCenter`, I delete the current `poolCenter` `hostCenter` to be sure it won't be re-elected because in this case the pool has no more hosts available.

In the other cases its okay to re-elect the same pool.